src/agent/session.rs

```diff
@@ -424,9 +425,17 @@ pub fn available_tools(
         tools.push(tools::RunCoverage::new(coverage_command).boxed());
     }
 
+    // reset file tool
     if let Some(env) = agent_env {
         tools.push(tools::ResetFile::new(&env.start_ref).boxed());
     }
 
+    tools.retain(|tool| {
+        !repository
+            .config()
+            .disabled_tools
+            .contains(&tool.name().to_string())
```

Comment:
nit, you shouldn't need to call to string. It's a Cow, try `tool.name().as_ref()`